titiler/pgstac/model.py

```diff
@@ -129,3 +173,9 @@ def validate_bbox(cls, v: BBox):
                 raise ValueError("Bounding box must be within (-180, -90, 180, 90)")
 
         return v
+
+
+class RegisterMosaic(PgSTACSearch):
+    """Model of /register endpoint input."""
+
+    metadata: Metadata = Field(default_factory=Metadata)
```

Comment:
Metadata will always be set when registering a `search` with the `RegisterMosaic`. default metadata should be `{"type": "mosaic"}`